src/CMakeLists.txt

```diff
@@ -0,0 +1,82 @@
+set(SRC glacierapp.cpp)
+set(HEADERS glacierapp.h)
+
+add_library(glacierapp SHARED ${SRC})
+add_library(Glacier::App ALIAS glacierapp)
+
+target_link_libraries(${PROJECT_NAME} PUBLIC
+	Qt5::Gui
+	Qt5::Qml
+	Qt5::Quick)
+
+if(MLITE5_FOUND)
+#	include_directories(${MLITE5_INCLUDE_DIRS})
+	add_definitions(-DHAS_MLITE5)
+	target_link_libraries(glacierapp PRIVATE
+		PkgConfig::MLITE5)
+endif()
+
+if(BOOSTABLE_FOUND)
+#	include_directories(${BOOSTABLE_INCLUDE_DIRS})
+	add_definitions(-DHAVE_CACHE=1)
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC
+	-I/usr/include/mdeclarativecache5")
```

Comment:
With CMake you don't have and should not provide include paths manually.
I found that `/usr/include/mdeclarativecache5` is a part of qdeclarative5-boostable and the target imported from pkg-config should already include this flag:
https://git.sailfishos.org/mer-core/mapplauncherd-qt/blob/master/data/pkgconfig/qdeclarative5-boostable.pc#L5
I think you can safely drop this flag here. `target_link_libraries(... PkgConfig::BOOSTABLE)` does the job.

I'm a bit of sorry to say those nitpicks that maybe discouraging developers from contribution and make you think like "why would I ever take such task? :disappointed: "
Reworking a build system (and packaging) is often underestimated. You'll have issues here, and there, and also elsewhere, and also on this platform with such position of the stars on the sky. To do it actually right you have to recursively fix all dependencies which have their own authors. Or if your scope is limited then you can make it look like the dependencies are fixed by providing your own CMake find modules (e.g. KDE use [those](https://cgit.kde.org/extra-cmake-modules.git/tree/find-modules) to make their CMakeLists look right).

The task is huge, you already did a very good job and I think that those nitpicks are the last. It makes sense to merge this as-is. Thank you for making the stuff better.

Reply:
No problem, I'm very glad that you thoroughly review it! I'm not put off quickly, don't you worry :wink: I'm taking this chance to learn CMake and make a start into C/C++ programming, so I rather do it good the first time around!